src/Client.js

```diff
@@ -11,23 +13,61 @@ function sanitizeAPIKey(key) {
   return key === '' ? null : key
 }
 
+function getTimeoutLength(timeout = 0, timeoutMaxAllowed) {
+  return (timeout - timeoutMaxAllowed) < 0 ? timeout : timeoutMaxAllowed
+}
+
 export default class Client {
   constructor(config) {
     this.config = Object.assign({}, defaults, config)
   }
 
   recognizeURL(url, options = {}) {
-    return this.connection(true, options)
-      .sendJson('/remote/recognitions', { url })
-      .then(this.handleResponse)
+    return this.recognizeRequest(
+      (options) => this.connection(true, options).sendJson('/remote/recognitions', { url }),
+      options
+    )
   }
 
   recognizeImage(data, options = {}) {
-    const params = {}
-    if (options.public) { params.public = true }
-    return this.connection(true, Object.assign({}, options, { params }))
-      .sendBinary('/recognitions', data)
-      .then(this.handleResponse)
+    const params = { public: options.public }
+    const fullOptions = Object.assign({}, options, { params })
+
+    return this.recognizeRequest(
+      (options) => this.connection(true, options).sendBinary('/recognitions', data),
+      fullOptions
+    )
+  }
+
+  // Takes a request and timeout and checks if the recognize request
+  // should start the polling process and calls poll if positive
+  recognizeRequest(requestFunc, options) {
+    if (options.timeout > 100) {
+      throw new ScnnrError('Timeout time greater than 100 not allowed')
```

Comment:
What is your intention of the 100 seconds limitation?
For example, `scnnr-ruby` currently allowed `Float::INFINITY` as `timeout`. Should we deny such as case?

Reply:
Sorry, I misunderstood that there was a limitation to the timeout value.

I can remove this, although I think having some limitation in place could be beneficial if for some reason the recognition takes really long or never resolves properly. Could leave the client polling forever if the user does not implement their own limit.

Reply:
Thank you for telling your thought! And I feel it's a good point.

I previously discussed it with the team. And the conclusion in that time is that when the user passes the option `{ timeout: Infinity }`, the user wants to wait forever (though actually, it's limited). In addition, we have no basis of the maximum value.
So we concluded that we should respect the value that the user passes.

What do you feel it? If you agree with it, could you remove it?